### ISSUE TITLE
Dialogs can remove accessibility listeners twice

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/joist#415]
@phet-steele showed me that many Dialog's are throwing an assertion that the dialog is trying to remove an accessibility listener that ins't in the list of listeners when the dialog is hidden.

The accessibility listeners were adaded in the Dialog constructor.  Since Dialog's are essentially disposed when hidden, we need to add the accessibility listeners when the Dialog is shown. (The update layout listener is added in this way).

**Comment by phet-steele** [comment]
Example error from GAO:
```
Error: Assertion failed: accessibleInput was not added
    at window.assertions.assertFunction (https://bayes.colorado.edu/continuous-testing/snapshot-1490127685046/assert/js/assert.js:21:13)
    at OptionsDialog.removeAccessibleInputListener (https://bayes.colorado.edu/continuous-testing/snapshot-1490127685046/scenery/js/accessibility/Accessibility.js?bust=1490127914271:378:21)
    at OptionsDialog.Dialog.removeAccessibleListeners (https://bayes.colorado.edu/continuous-testing/snapshot-1490127685046/joist/js/Dialog.js?bust=1490127914271:190:12)
    at OptionsDialog.hide (https://bayes.colorado.edu/continuous-testing/snapshot-1490127685046/joist/js/Dialog.js?bust=1490127914271:237:14)
    at Object.fire (https://bayes.colorado.edu/continuous-testing/snapshot-1490127685046/scenery-phet/js/BarrierRectangle.js?bust=1490127914271:48:57)
    at ButtonListener.setButtonState (https://bayes.colorado.edu/continuous-testing/snapshot-1490127685046/scenery/js/input/ButtonListener.js?bust=1490127914271:129:31)
    at Object.up (https://bayes.colorado.edu/continuous-testing/snapshot-1490127685046/scenery/js/input/ButtonListener.js?bust=1490127914271:87:14)
    at ButtonListener.buttonUp (https://bayes.colorado.edu/continuous-testing/snapshot-1490127685046/scenery/js/input/DownUpListener.js?bust=1490127914271:123:22)
    at Object.up (https://bayes.colorado.edu/continuous-testing/snapshot-1490127685046/scenery/js/input/DownUpListener.js?bust=1490127914271:56:16)
    at Input.dispatchToPointer (https://bayes.colorado.edu/continuous-testing/snapshot-1490127685046/scenery/js/input/Input.js?bust=1490127914271:773:27)
Approximately 3/21/2017, 2:21:25 PM
```

**Comment by jessegreenberg** [comment]
Thanks @phet-steele, should be fixed in the above commit. closing, but let me know if you still run into this.
